src/lustre/typeCheckerContext.ml

```diff
@@ -134,7 +141,7 @@ let rec lookup_ty_syn: tc_context -> LA.ident -> tc_type option
                   then Some ty
                   else lookup_ty_syn ctx uid
                | _ -> Some ty )
-  | None -> None
+  | None -> IMap.find_opt i (ctx.ty_syns)
```

Comment:
Something is off. You are repeating the exact same search.

Reply:
Addressed